src/neo/Ledger/Blockchain.cs

```diff
@@ -266,19 +268,30 @@ private void OnImport(IEnumerable<Block> blocks, bool verify)
         private void AddUnverifiedBlockToCache(Block block)
         {
             // Check if any block proposal for height `block.Index` exists
-            if (!block_cache_unverified.TryGetValue(block.Index, out LinkedList<Block> blocks))
+            if (!block_cache_unverified.TryGetValue(block.Index, out var blocks))
             {
                 // There are no blocks, a new LinkedList is created and, consequently, the current block is added to the list
-                blocks = new LinkedList<Block>();
+                blocks = new UnverifiedBlocks();
                 block_cache_unverified.Add(block.Index, blocks);
             }
-            // Check if any block with the hash being added already exists on possible candidates to be processed
-            foreach (var unverifiedBlock in blocks)
+            else
             {
-                if (block.Hash == unverifiedBlock.Hash)
+                // Check if any block with the hash being added already exists on possible candidates to be processed
+                foreach (var unverifiedBlock in blocks.Blocks)
+                {
+                    if (block.Hash == unverifiedBlock.Hash)
+                        return;
+                }
+
+                if (!blocks.Nodes.Add(Sender))
```

Comment:
You can't use `Sender` because the message is sent by `NoSender` in `RemoteNode`.

Reply:
What happens if you send a message to dead actor? In this case we do not send a message, but we access to it...

Reply:
> Can we remove , ActorRefs.NoSender in RemoteNode?

Yes.

> What happens if you send a message to dead actor?

Nothing happens.

Reply:
done